charts/redpanda/templates/secrets.yaml

```
@@ -80,9 +80,11 @@ stringData:
       touch /tmp/postStartHookFinished
     }
 
+  {{- if not ( dig "node" "recovery_mode_enabled" false .Values.config ) }}
```

Should this condition have something like
```suggestion
  {{- if and (not ( dig "node" "recovery_mode_enabled" false .Values.config )) (include "redpanda-atleast-23-3-1" $root | fromJson).bool) }}
```

---

Good point, but honestly RP will complain about not knowing this configuration before we hit this.